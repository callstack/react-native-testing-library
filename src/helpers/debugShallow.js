// @flow
import * as React from 'react';
import { shallowInternal } from '../shallow';
import format from './format';

/**
 * Log pretty-printed shallow test component instance
 */
export default function debugShallow(
  instance: ReactTestInstance | React.Element<any>,
  message?: string
) {
  const { output } = shallowInternal(instance);

  if (message) {
    console.log(`${message}\n\n`, format(output));
  } else {
    console.log(format(output));
  }
}
