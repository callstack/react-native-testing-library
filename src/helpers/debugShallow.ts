import * as React from 'react';
import type { ReactTestInstance } from 'react-test-renderer';
import { shallowInternal } from '../shallow';
import format from './format';

/**
 * Log pretty-printed shallow test component instance
 */
export default function debugShallow(
  instance: ReactTestInstance | React.ReactElement<any>,
  message?: string
) {
  const { output } = shallowInternal(instance);

  if (message) {
    console.log(`${message}\n\n`, format(output));
  } else {
    console.log(format(output));
  }
}
