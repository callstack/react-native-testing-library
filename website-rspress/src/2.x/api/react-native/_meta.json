[
  {
    "type": "file",
    "name": "index",
    "label": "Table of Contents"
  },
  {
    "type": "dir",
    "name": "classes",
    "label": "Classes"
  },
  {
    "type": "dir",
    "name": "interfaces",
    "label": "Interfaces"
  },
  {
    "type": "dir",
    "name": "types",
    "label": "Types"
  }
]
