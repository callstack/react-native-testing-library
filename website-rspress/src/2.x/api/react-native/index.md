# Re.Pack React Native APIs

## Table of contents

### Classes

- [Chunk](./classes/Chunk.md)
- [ChunkManager](./classes/ChunkManager.md)

### Interfaces

- [ChunkConfig](./interfaces/ChunkConfig.md)
- [ChunkManagerConfig](./interfaces/ChunkManagerConfig.md)
- [RemoteChunkLocation](./interfaces/RemoteChunkLocation.md)
- [StorageApi](./interfaces/StorageApi.md)

### Type aliases

- [RemoteChunkResolver](./types/RemoteChunkResolver.md)
