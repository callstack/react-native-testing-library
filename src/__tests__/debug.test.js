// @flow
/* eslint-disable no-console */
import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import stripAnsi from 'strip-ansi';
import { debug, render, fireEvent, flushMicrotasksQueue } from '..';
import debugShallow from '../helpers/debugShallow';

type ConsoleLogMock = JestMockFn<Array<string>, void>;

function TextComponent({ text }) {
  return <Text>{text}</Text>;
}

class Button extends React.Component<any, any> {
  state = { counter: 0 };

  onPress = async () => {
    await Promise.resolve();

    this.setState({ counter: 1 });
    this.props.onPress();
  };

  render() {
    return (
      <TouchableOpacity onPress={this.onPress}>
        <TextComponent text={`${this.props.text} ${this.state.counter}`} />
      </TouchableOpacity>
    );
  }
}

test('debug', () => {
  jest.spyOn(console, 'log').mockImplementation(x => x);
  const component = <Button onPress={jest.fn} text="Press me" />;
  debug(component);

  const output = ((console.log: any): ConsoleLogMock).mock.calls[0][0];

  expect(stripAnsi(output)).not.toEqual(output);
  expect(stripAnsi(output)).toMatchSnapshot();

  ((console.log: any): ConsoleLogMock).mockReset();

  debug(component, 'test message');

  expect(console.log).toHaveBeenCalledWith('test message\n\n', output);
});

test('debug.shallow', () => {
  expect(debug.shallow).toBe(debugShallow);
});

test('debug.deep', () => {
  // $FlowFixMe
  console.log = jest.fn();
  const component = <Button onPress={jest.fn} text="Press me" />;
  debug.deep(component);

  const output = ((console.log: any): ConsoleLogMock).mock.calls[0][0];

  expect(stripAnsi(output)).not.toEqual(output);
  expect(stripAnsi(output)).toMatchSnapshot();

  ((console.log: any): ConsoleLogMock).mockReset();

  debug.deep(component, 'test message');

  expect(console.log).toHaveBeenCalledWith('test message\n\n', output);
});

test('debug.deep async test', async () => {
  // $FlowFixMe
  console.log = jest.fn();
  const { toJSON, getByName } = render(
    <Button onPress={jest.fn} text="Press me" />
  );

  fireEvent.press(getByName('TouchableOpacity'));
  await flushMicrotasksQueue();

  debug.deep(toJSON());

  const output = ((console.log: any): ConsoleLogMock).mock.calls[0][0];

  expect(stripAnsi(output)).toMatchSnapshot();
});
