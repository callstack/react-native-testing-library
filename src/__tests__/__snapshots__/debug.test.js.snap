// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debug 1`] = `
"<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function _callee]}
>
  <TextComponent
    text=\\"Press me 0\\"
  />
</TouchableOpacity>"
`;

exports[`debug.deep 1`] = `
"<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function bound touchableHandlePress]}
  onResponderGrant={[Function bound touchableHandleResponderGrant]}
  onResponderMove={[Function bound touchableHandleResponderMove]}
  onResponderRelease={[Function bound touchableHandleResponderRelease]}
  onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
  onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
  onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
  style={
    Object {
      \\"opacity\\": 1,
    }
  }
>
  <Text>
    Press me 0
  </Text>
</View>"
`;

exports[`debug.deep async test 1`] = `
"<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function bound touchableHandlePress]}
  onResponderGrant={[Function bound touchableHandleResponderGrant]}
  onResponderMove={[Function bound touchableHandleResponderMove]}
  onResponderRelease={[Function bound touchableHandleResponderRelease]}
  onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
  onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
  onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
  style={
    Object {
      \\"opacity\\": 1,
    }
  }
>
  <Text>
    Press me 1
  </Text>
</View>"
`;
